acceptance/tests/facts/partitions.rb

```diff
@@ -0,0 +1,37 @@
+test_name "C96148: verify partitions facts" do
+  tag 'risk:high'
+
+  confine :except, :platform => 'osx' # no partitions on osx
+  confine :except, :platform => 'windows' # no partitions on windows
+  confine :except, :platform => 'solaris' # no partitions on solaris
+
+  require 'json'
+
+  possible_facts = [
```

Comment:
Because when I started writing it as a bunch of "unless .nil? ... end" blocks it made sense to refacter it into an array and use a common loop over the ones that might or might not be set to reduce the code.  I can make it a hash if you'd prefer.

I only put the other code in base_fact_util.rb because of how large it was. Given that this tests all fits on a screen, and no one else is going to needs these facts, I'd ready see it in the test so you have one place to look.